scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,227 @@
+        for variable in sparql_map['fields']:
+            variables += '?' + variable + ' '
+        pattern = sparql_map['pattern'].format(uri)
+        sparql_query = self.QUERY_TEMPLATE.format(variables, pattern)
+        self.sparql_wrapper.setQuery(sparql_query)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        ret = []
+        for result in results['results']['bindings']:
+            item = {}
+            for variable in sparql_map['fields']:
+                item[variable] = result[variable]['value']
+            ret.append(item)
+        return ret
+
+    def get_records(self, uris, sparql_mapping):
+        records = []
+        for uri in uris:
+            record = {}
+            record['uri'] = uri
+            for sparql_map in sparql_mapping:
+                if sparql_map['type'] == 'string':
+                    record[sparql_map['name']] = self.get_string(uri, sparql_map)
+                if sparql_map['type'] == 'array':
+                    record[sparql_map['name']] = self.get_array(uri, sparql_map)
+                if sparql_map['type'] == 'dict':
+                    record[sparql_map['name']] = self.get_dict(uri, sparql_map)
+            record['authors'] = self.complete_authors(record['authors'])
+            records.append(record)
+        return records
+
+    def get_total(self, start_date, end_date):
+        query_str = self.GET_TOTAL_QUERY.format(start_date.isoformat(), end_date.isoformat())
+        self.sparql_wrapper.setQuery(query_str)
+        result = self.sparql_wrapper.query()
+        result = result.convert()
+        return int(result['results']['bindings'][0]['total']['value'])
+
+    def get_uris(self, start_date, end_date, limit, offset):
+        query_str = self.GET_URIS_QUERY.format(start_date.isoformat(), end_date.isoformat(), limit, offset)
+        self.sparql_wrapper.setQuery(query_str)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        return [result['uri']['value'] for result in results['results']['bindings']]
+
+    def complete_authors(self, authors):
+        for author in authors:
+            email = self.get_string(author['sameAs'], mapping.AUTHOR_MAPPING['email'])
+            if email:
+                author['email'] = email
+            affiliation = self.get_dict(author['sameAs'], mapping.AUTHOR_MAPPING['affiliation'])
+            if affiliation:
+                author['affiliation'] = affiliation
+            orcidId = self.get_string(author['sameAs'], mapping.AUTHOR_MAPPING['orcidId'])
+            author['sameAs'] = [author['sameAs']]
+            if orcidId:
+                author['sameAs'].append(orcidId)
+        return authors
+
+    @property
+    def schema(self):
+        return {
+            'title': ('/title', lambda x: x if x else ''),
+            'providerUpdatedDateTime': ('/date', lambda x: datetime.strptime(x, "%Y-%m-%d").strftime("%Y-%m-%dT%H:%M:%S%Z") + '+00:00'),
```

Comment:
Oh, just noticed this. We actually have a helper function (`helpers.datetime_formatter`), which ensures that all datetimes we receive are processed into an identical format. could you use that instead?